src/GUI/EFCorePowerTools/Dialogs/PickServerDatabaseDialog.xaml.cs

```diff
@@ -1,97 +1,74 @@
+    using Shared.Models;
 
-        private void SaveButton_Click(object sender, RoutedEventArgs e)
-        {
-            DialogResult = true;
-            Close();
-        }
+    public partial class PickServerDatabaseDialog : IPickServerDatabaseDialog
+    {
+        private readonly Func<(DatabaseConnectionModel Connection, DatabaseDefinitionModel Definition)> _getDialogResult;
+        private readonly Action<IEnumerable<DatabaseConnectionModel>> _addConnections;
+        private readonly Action<IEnumerable<DatabaseDefinitionModel>> _addDefinitions;
 
-        private void CancelButton_Click(object sender, RoutedEventArgs e)
+        public PickServerDatabaseDialog(ITelemetryAccess telemetryAccess,
+                                        IPickServerDatabaseViewModel viewModel)
         {
-            DialogResult = false;
-            Close();
-        }
+            telemetryAccess.TrackPageView(nameof(PickServerDatabaseDialog));
 
-        private void comboBox1_SelectionChanged(object sender, System.Windows.Controls.SelectionChangedEventArgs e)
-        {
-            if (comboBox1.SelectedItem != null)
+            DataContext = viewModel;
+            viewModel.CloseRequested += (sender, args) =>
             {
-                SelectedDatabase = (KeyValuePair<string, DatabaseInfo>)comboBox1.SelectedItem;
-                DacpacPath = null;
-                comboBox2.SelectedIndex = -1;
+                DialogResult = args.DialogResult;
+                Close();
             };
-        }
-
-        private void comboBox2_SelectionChanged(object sender, System.Windows.Controls.SelectionChangedEventArgs e)
-        {
-            if (comboBox2.SelectedItem != null)
+            _getDialogResult = () => (viewModel.SelectedDatabaseConnection, viewModel.SelectedDatabaseDefinition);
+            _addConnections = models =>
+            {
+                foreach (var model in models)
+                    viewModel.DatabaseConnections.Add(model);
+            };
+            _addDefinitions = models =>
             {
-                var result = (KeyValuePair<string, string>)comboBox2.SelectedItem;
-                DacpacPath = result.Value;
-                comboBox1.SelectedIndex = -1;
+                foreach (var model in models)
+                    viewModel.DatabaseDefinitions.Add(model);
             };
+
+            InitializeComponent();
         }
 
         private void Window_Loaded(object sender, RoutedEventArgs e)
         {
-            comboBox1.Focus();
+            DatabaseConnectionCombobox.Focus();
         }
 
-        private void AddButton_OnClick(object sender, RoutedEventArgs e)
+        public (bool ClosedByOK, (DatabaseConnectionModel Connection, DatabaseDefinitionModel Definition) Payload) ShowAndAwaitUserResponse(bool modal)
         {
-            try
-            {
-                var info = EnvDteHelper.PromptForInfo(_package);
-                if (info.DatabaseType == DatabaseType.SQLCE35) return;
```

Comment:
@ErikEJ Regarding the return unknown, do you mean this?
The old cold just returned, if it's `SQLCE35`.

If not, please explain a little bit more what you mean by "return unknown" on the new implementation 😄

Reply:
Make PromptForInfo return DatabasteType.Unknown instead of SQLCE35